src/mlpack/methods/ann/layer/upsample_impl.hpp

```diff
@@ -0,0 +1,275 @@
+/**
+ * @file methods/ann/layer/upsample_impl.hpp
+ * @author Abhinav Anand
+ *
+ * Implementation of the Upsample layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_UPSAMPLE_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_UPSAMPLE_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "upsample.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+
+template<typename InputDataType, typename OutputDataType>
+  Upsample<InputDataType, OutputDataType>::
```

Comment:
```suggestion
Upsample<InputDataType, OutputDataType>::
```

There are some other places with the same issue.